framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -219,26 +219,31 @@ protected BaseTypeVisitor(BaseTypeChecker checker, Factory typeFactory) {
      */
     @SuppressWarnings("unchecked") // unchecked cast to type variable
     protected Factory createTypeFactory() {
-        // Try to reflectively load the type factory.
-        Class<?> checkerClass = checker.getClass();
-        while (checkerClass != BaseTypeChecker.class) {
-            final String classToLoad =
-                    checkerClass
-                            .getName()
-                            .replace("Checker", "AnnotatedTypeFactory")
-                            .replace("Subchecker", "AnnotatedTypeFactory");
-
-            AnnotatedTypeFactory result =
-                    BaseTypeChecker.invokeConstructorFor(
-                            classToLoad,
-                            new Class<?>[] {BaseTypeChecker.class},
-                            new Object[] {checker});
-            if (result != null) {
-                return (Factory) result;
-            }
-            checkerClass = checkerClass.getSuperclass();
-        }
-        return (Factory) new BaseAnnotatedTypeFactory(checker);
+        try {
+            // Try to reflectively load the type factory.
+            Class<?> checkerClass = checker.getClass();
+            while (checkerClass != BaseTypeChecker.class) {
+                final String classToLoad =
+                        checkerClass
+                                .getName()
+                                .replace("Checker", "AnnotatedTypeFactory")
+                                .replace("Subchecker", "AnnotatedTypeFactory");
+
+                AnnotatedTypeFactory result =
+                        BaseTypeChecker.invokeConstructorFor(
+                                classToLoad,
+                                new Class<?>[] {BaseTypeChecker.class},
+                                new Object[] {checker});
+                if (result != null) {
+                    return (Factory) result;
+                }
+                checkerClass = checkerClass.getSuperclass();
+            }
+            return (Factory) new BaseAnnotatedTypeFactory(checker);
+        } catch (Throwable t) {
```

Comment:
The only place that could still throw an unexpected exception is the call to `BaseAnnotatedTypeFactory`. Please limit the scope of the try block to that, to prevent duplicate errors when a subclass checker is executed on line 233.